R/literal_coercion_linter.R

```diff
@@ -12,23 +12,41 @@
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
 literal_coercion_linter <- function() {
-  coercers <- xp_text_in_table(paste0(
-    "as.",
-    c("logical", "integer", "numeric", "double", "character")
-  ))
+  rlang_coercers <- xp_text_in_table(
+    c("lgl", "int", "dbl", "chr")
+  )
+  base_coercers <- xp_text_in_table(
+      paste0("as.", c("logical", "integer", "numeric", "double", "character"))
+  )
   # notes for clarification:
   #  - as.integer(1e6) is arguably easier to read than 1000000L
   #  - in x$"abc", the "abc" STR_CONST is at the top level, so exclude OP-DOLLAR
   #  - need condition against STR_CONST w/ EQ_SUB to skip quoted keyword arguments (see tests)
+  #  - for {rlang} coercers, both `int(1)` and `int(1, )` need to be linted
   xpath <- glue::glue("//expr[
-    expr[1][SYMBOL_FUNCTION_CALL[ {coercers} ]]
-    and expr[2][
-      not(OP-DOLLAR)
-      and (
-        NUM_CONST[not(contains(translate(text(), 'E', 'e'), 'e'))]
-        or STR_CONST[not(following-sibling::*[1][self::EQ_SUB])]
-      )
-    ]
+    (
+      expr[1][SYMBOL_FUNCTION_CALL[ {base_coercers} ]]
+      and expr[2][
+        not(OP-DOLLAR)
+        and (
+          NUM_CONST[not(contains(translate(text(), 'E', 'e'), 'e'))]
+          or STR_CONST[not(following-sibling::*[1][self::EQ_SUB])]
+        )
+      ]
+    ) or
+    (
+        (
+          expr[1][SYMBOL_FUNCTION_CALL[ {rlang_coercers} ]]
+          and expr[2][
+          not(OP-DOLLAR)
```

Comment:
probably makes sense to save this condition as a variable and `glue()` it in since it's reused

Reply:
Thanks! Simplifies the path significantly. Done!